packages/flutter/lib/src/widgets/scroll_physics.dart

```diff
@@ -428,18 +428,60 @@ class RangeMaintainingScrollPhysics extends ScrollPhysics {
     @required bool isScrolling,
     @required double velocity,
   }) {
-    if (velocity != 0.0 || ((oldPosition.minScrollExtent == newPosition.minScrollExtent) && (oldPosition.maxScrollExtent == newPosition.maxScrollExtent))) {
-      return super.adjustPositionForNewDimensions(oldPosition: oldPosition, newPosition: newPosition, isScrolling: isScrolling, velocity: velocity);
+    bool maintainOverscroll = true;
```

Comment:
Maybe we should update the docs on `RangeMaintainingScrollPhysics` indicate that there are more cases where the range is not maintained.